doc/migrate-to-pymongo4.rst

```diff
@@ -121,6 +121,27 @@ can be changed to this::
 Database
 --------
 
+Database.authenticate and Database.logout are removed
+.....................................................
+
+Removed :meth:`pymongo.database.Database.authenticate` and
+:meth:`pymongo.database.Database.logout`. Authenticating multiple users
+on the same client conflicts with support for logical sessions in MongoDB 3.6+.
+To authenticate as multiple users, create multiple instances of
+:class:`~pymongo.mongo_client.MongoClient`. Code like this::
+
+    client = MongoClient()
+    client.admin.authenticate('user1', 'pass1')
+    client.admin.authenticate('user2', 'pass2')
+
+can be changed to this::
+
+    client1 = MongoClient(username='user1', password='pass1')
+    client2 = MongoClient(username='user2', password='pass2')
+
+Alternatively, create a single user that contains all the authentication privileges
```

Comment:
This sounds like a much better pattern. Do you think it is feasible for us to say loudly that this is what we **recommend**?

Reply:
I think both options are valid. Sometimes you would want to use a separate client entirely, for example if the app needs to run maintenance tasks in addition to regular CRUD. It would not be ideal (security-wise) to give the entire CRUD app all the maintenance task's privileges. In the security world this is called the principle of least privilege: https://en.wikipedia.org/wiki/Principle_of_least_privilege